tooling/noir_js/src/program.ts

```diff
@@ -1,13 +1,14 @@
 /* eslint-disable  @typescript-eslint/no-explicit-any */
 import { Backend, CompiledCircuit, ProofData } from '@noir-lang/types';
 import { generateWitness } from './witness_generation.js';
-import initAbi from '@noir-lang/noirc_abi';
+import initAbi, { abiDecode, InputMap, InputValue } from '@noir-lang/noirc_abi';
 import initACVM from '@noir-lang/acvm_js';
+import { witnessMapToUint8Array } from './serialize.js';
 
 export class Noir {
   constructor(
     private circuit: CompiledCircuit,
-    private backend: Backend,
+    private backend?: Backend,
```

Comment:
marked as optional incase the user just wants to execute

Reply:
Yep, that's correct